src/clj/rems/api/entitlements.clj

```diff
@@ -20,4 +21,14 @@
                      {resource :- (describe s/Str "return entitlements for this resource (optional)") nil}
                      {expired :- (describe s/Bool "whether to include expired entitlements") false}]
       :return GetEntitlementsResponse
-      (ok (entitlements/get-entitlements-for-api user resource expired)))))
+      (ok (get-entitlements-for-api {:user-id user
+                                     :resource-ext-id resource
+                                     :expired expired})))
+
+    (GET "/csv" []
+      :summary "Return entitlements as CSV"
+      :roles #{:handler :reporter}
+      :produces ["text/csv"]
+      :responses {200 {:schema s/Str}}
+      (-> (ok (get-entitlements-for-csv-export))
+          (response/content-type "text/csv")))))
```

Comment:
idea sounds good, but as you said, the format is currently different to the base route entitlements, and roles are different too (csv export only allows handler and reporter roles). we could change csv export to work with the same attributes and roles. we could figure out proper `format` from request headers and/or query params. documenting the API with swagger doesn't work very well, as swagger 2.0 doesn't seem to support `x/oneOf`, so we cannot document different schemas for same response code.

maybe best way forward would be to refactor underlying code so that both JSON and CSV routes use same functions and differ only in response format? if/when we make the move to a later version of swagger, then we can use the multiple schemas responses with relative ease.